packages/react-form-renderer/src/form-renderer/form-renderer.js

```diff
@@ -1,77 +1,132 @@
-import React, { useState, useRef } from 'react';
-import Form from '../form';
 import arrayMutators from 'final-form-arrays';
-import PropTypes from 'prop-types';
 import createFocusDecorator from 'final-form-focus';
+import PropTypes from 'prop-types';
+import React, {useCallback, useMemo, useRef, useState} from 'react';
 
+import defaultSchemaValidator from '../default-schema-validator';
+import defaultValidatorMapper from '../validator-mapper';
+import Form from '../form';
 import RendererContext from '../renderer-context';
 import renderForm from './render-form';
-import defaultSchemaValidator from '../default-schema-validator';
 import SchemaErrorComponent from './schema-error-component';
-import defaultValidatorMapper from '../validator-mapper';
+
+
+const isFunc = (fn) => (typeof fn === 'function');
 
 const FormRenderer = ({
+  actionMapper,
+  children,
+  clearedValue,
+  clearOnUnmount,
   componentMapper,
+  decorators,
   FormTemplate,
-  onSubmit,
+  FormTemplateProps,
+  mutators,
   onCancel,
+  onError,
   onReset,
-  clearOnUnmount,
-  subscription,
-  clearedValue,
+  onSubmit,
   schema,
-  validatorMapper,
-  actionMapper,
   schemaValidatorMapper,
+  subscription,
+  validatorMapper,
   ...props
 }) => {
   const [fileInputs, setFileInputs] = useState([]);
   const registeredFields = useRef({});
   const focusDecorator = useRef(createFocusDecorator());
-  let schemaError;
+  const validatorMapperMerged = useMemo(() => ({
+    ...defaultValidatorMapper,
+    ...validatorMapper
+  }), [validatorMapper]);
```

Comment:
Not sure if dependency array will work well with an object, the same applies for mutators/decorators (arrays)